modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/service_base_impl.ftl

```diff
@@ -537,6 +631,30 @@ import org.osgi.service.component.annotations.Reference;
 					return ${entity.variableName}Persistence.findBy${entity.externalReferenceCode?cap_first[0..0]}_ERC(${entity.externalReferenceCode}Id, externalReferenceCode);
 				}
 			</#if>
+
+			<#if entity.hasLazyReference() && entity.hasEntityColumn("companyId") && (!entity.hasEntityColumn("groupId") || stringUtil.equals(entity.name, "Group"))>
+				@Override
+				public List<${entity.name}> getBatchImported${entity.pluralName}(long companyId, int batchImportStatus, int start, int end) <#if (serviceBaseExceptions?size gt 0)>throws ${stringUtil.merge(serviceBaseExceptions)} </#if>{
+					return ${entity.variableName}Persistence.findByC_BIS(companyId, batchImportStatus, start, end);
+				}
+
+				@Override
+				public int getBatchImported${entity.pluralName}Count(long companyId, int batchImportStatus) <#if (serviceBaseExceptions?size gt 0)>throws ${stringUtil.merge(serviceBaseExceptions)} </#if>{
+					return ${entity.variableName}Persistence.countByC_BIS(companyId, batchImportStatus);
+				}
+			</#if>
+
+			<#if entity.hasLazyReference() && entity.hasEntityColumn("groupId") && !stringUtil.equals(entity.name, "Group")>
```

Comment:
don't we have to check also for the hasExternalReferenceCode?

Reply:
Same here